pkg/nsx/services/subnet/builder.go

```diff
@@ -0,0 +1,109 @@
+package subnet
+
+import (
+	"fmt"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var (
+	String = common.String
+	Int64  = common.Int64
+	Bool   = common.Bool
+)
+
+func getCluster(service *SubnetService) string {
+	return service.NSXConfig.Cluster
+}
+
+func (service *SubnetService) buildSubnet(obj client.Object, tags []model.Tag) (*model.VpcSubnet, error) {
+	tags = append(tags, service.buildBasicTags(obj)...)
```

Comment:
we also need to add NS labels to NSX subnet tags, do you plan to support that in this patch or following patch?

Reply:
I think the interface is ready, will do this in following patch.